engine/install/debian.md

```diff
@@ -84,27 +84,14 @@ from the repository.
         apt-transport-https \
         ca-certificates \
         curl \
-        gnupg-agent \
-        software-properties-common
+        gnupg-agent
     ```
 
 2.  Add Docker's official GPG key:
 
     ```bash
-    $ curl -fsSL {{ download-url-base }}/gpg | sudo apt-key add -
-    ```
-
-    Verify that you now have the key with the fingerprint
-    `9DC8 5822 9FC7 DD38 854A  E2D8 8D81 803C 0EBF CD88`, by searching for the
-    last 8 characters of the fingerprint.
-
-    ```bash
-    $ sudo apt-key fingerprint 0EBFCD88
+    $ curl -fsSL {{ download-url-base }}/gpg | gpg --dearmor | sudo tee /usr/share/keyrings/docker-ce-archive-keyring.gpg > /dev/null
```

Comment:
Wondering if we need some step after this to (manually) verify the fingerprint

Reply:
If we add `.asc` to the filename, we don't need to `--dearmor` :sweat_smile:

Although that makes it slightly more difficult to add a verification step. :thinking:

(I personally use `gpg --keyserver ha.pool.sks-keyservers.net --recv-keys 9DC858229FC7DD38854AE2D88D81803C0EBFCD88` to fetch this key, which directly verifies it, but it's then much less reliable in fetching, ala https://github.com/docker-library/faq#openpgp--gnupg-keys-and-verification)

Maybe it would be enough to list the fingerprint here and leave it as an exercise for the reader to determine how to verify it?  We could provide the hint of `gpg --import` + `gpg --fingerprint` which are the two commands that will help them do so.

I also wonder whether `/usr/share/keyrings/...` is the "right" place for us to suggest users be downloading this file.  I don't know that there's really an "official" place for sideloaded keyrings like this (I personally use `/etc/apt/tianon.gpg.d` :sweat_smile:), but `/usr/share` is traditionally supposed to be filled by the package manager (with `/usr/local/share` being the "user supplied" variant), so perhaps we instead suggest users create `/usr/local/share/keyrings` and put it there?

(Other than those, I'm a big fan of this PR!  :smile: :metal:)

Reply:
Where does `apt-key add` store it? Does that specify some "canonical" path where to store the information?

Reply:
It stores it inside `/etc/apt/trusted.gpg` - there exists an `/etc/apt/trusted.gpg.d/` that is an appropriate place for these things, but any keys added in either of those are trusted for *all* repo entries, which is why this added the additional `signed-by=` bit to the repo line (so that the key is only trusted for that one repo instead, which is much safer and the recommended solution by the APT and Debian GnuPG maintainers).

Reply:
> I also wonder whether `/usr/share/keyrings/...` is the "right" place

@tianon, it's where [Debian's own Wiki](https://wiki.debian.org/DebianRepository/UseThirdParty#OpenPGP_Key_distribution) suggests we put it :shrug: :)


Quote (emphasis theirs):
>> The key MUST be downloaded over a secure mechanism like HTTPS to a location only writable by root, which SHOULD be /usr/share/keyrings.

@thaJeztah, I believe `apt-key` appends them to the `/etc/apt/trusted.gpg` file

Reply:
> I believe apt-key appends them to the /etc/apt/trusted.gpg file

Ah, thx

Is the name of the file important, or would any name do?  Looking if `docker-ce-archive-keyring.gpg` is a good fit, or if we should use something else (FWIW, now that "Docker EE" is no longer a thing, we are toning down the "CE" moniker a bit; we won't be renaming packages as we know that's quite disruptive, but perhaps something more generic could work for the filename)

Reply:
> Is the name of the file important, or would any name do?

Any name will do both for the `.gpg` keyring and the `.list` file and they can be different from one another, so we can future-proof them with generic filenames as you wish :)

Reply:
`apt-key add` appends keys to `/etc/apt/trusted.gpg`, which is not used anymore by most modern distro versions. Usually keys are added to `/etc/apt/trusted.gpg.d/name.gpg`, which is queried by APT automatically. So I suggest to use `/etc/apt/trusted.gpg.d/docker.gpg` to simplify and align it with how Debian and Ubuntu do it on current stable and testing versions.

To avoid the `sudo tee` pipe, `gpg` actually has an output argument, so the following would be simpler:
```
curl -fsSL {{ download-url-base }}/gpg | sudo gpg --dearmor -o /etc/apt/trusted.gpg.d/docker.gpg --yes
```
`--yes` allows to overwrite an existing file, e.g. in case of an update. But probably this is better skipped to not risk overwriting other keys and in case let the admin decide about this interactively.

Reply:
Just for posterity, I clarified the text from https://wiki.debian.org/DebianRepository/UseThirdParty#OpenPGP_Key_distribution with some other Debian Developers and the consensus was that the intent there was likely that the user is creating a file which will almost immediately be overwritten (and thus then owned/managed) by a file provided by a `deriv-archive-keyring` package installed from that overlay, which is how most derivatives work and manage keyring updates in a similar way to how Debian itself does so.

That's not the case for Docker, but it's reasonably harmless to use the non-local path anyhow (the chance for conflicts on this filename is extremely low, and the minority of users who are dogmatic about the FHS will grumble as they read this and adjust the path for themselves anyhow :smile:). :+1: